apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -576,15 +546,13 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
           tag: "write_on_server_owned_interface"
         )
 
-        ask_clean_session(new_state)
-        MessageTracker.discard(new_state.message_tracker, message_id)
+        {:ok, updated_state} = ask_clean_session(new_state, message_id, timestamp)
```

Comment:
Just rebind `new_state`, imho having yet another name for a new version of the state does more harm than good (moreover, it introduces a lot of noise in the MR).